.github/workflows/continuous-integration.yml

```diff
@@ -0,0 +1,112 @@
+          discovery.type: single-node
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6.9
+    - name: Update pip
+      run: python -m pip install --upgrade pip
+    - name: Install tox
+      run: python -m pip install tox
+    - name: Checkout git repo
+      uses: actions/checkout@v2
+    - name: Create test databases
+      run: psql -U postgres -h localhost -p 5432 -c 'CREATE DATABASE htest'
+    - name: Cache the .tox dir
+      uses: actions/cache@v2
+      with:
+        path: |
+          .tox/checkformatting
+          .tox/tests
+          .tox/coverage
+        key: ${{ runner.os }}-tox-backend-${{ hashFiles('tox.ini') }}-${{ hashFiles('requirements*') }}-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }}
+        restore-keys: |
+          ${{ runner.os }}-tox-backend-
+    - name: Run tox
+      run: tox --parallel auto -e checkformatting,tests,coverage
+  functests:
+    runs-on: ubuntu-latest
+    services:
+      postgres:
+        image: postgres:11.5-alpine
+        ports:
+        - 5432:5432
+      elasticsearch:
+        image: hypothesis/elasticsearch:latest
+        ports:
+        - 9200:9200
+        env:
+          discovery.type: single-node
+      rabbitmq:
+        image: rabbitmq:3.6-management-alpine
+        ports:
+        - 5672:5672
+    steps:
+    - name: Set up Python
+      uses: actions/setup-python@v2
+      with:
+        python-version: 3.6.9
+    - name: Update pip
+      run: python -m pip install --upgrade pip
+    - name: Install tox
+      run: python -m pip install tox
+    - name: Checkout git repo
+      uses: actions/checkout@v2
+    - name: Create test databases
+      run: psql -U postgres -h localhost -p 5432 -c 'CREATE DATABASE htest'
+    - name: Cache the .tox dir
+      uses: actions/cache@v2
+      with:
+        path: |
+          .tox/lint
+          .tox/functests
+        key: ${{ runner.os }}-tox-functests-${{ hashFiles('tox.ini') }}-${{ hashFiles('requirements*') }}-${{ hashFiles('setup.py') }}-${{ hashFiles('setup.cfg') }}
+        restore-keys: |
+          ${{ runner.os }}-tox-functests-
+    - name: Cache the node_modules dir
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-node_modules-${{ hashFiles('package-lock.json') }}
+    - name: npm install
+      run: npm install
+    - name: gulp build
+      run: gulp build
+    - name: Run tox
+      run: tox --parallel auto -e lint,functests
+  frontend:
+    runs-on: ubuntu-latest
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Cache the node_modules dir
+      uses: actions/cache@v2
+      with:
+        path: node_modules
+        key: ${{ runner.os }}-node_modules-${{ hashFiles('package-lock.json') }}
+    - name: Install
+      run: npm install
+    - name: Format
+      run: npm run-script checkformatting
+    - name: Lint
+      run: npm run-script lint
```

Comment:
I think I would recommend running `make frontend-lint` here rather than `npm run-script`. The reason is that the frontend lint steps can vary slightly across projects and time and `make frontend-lint` will always run whatever is appropriate in the current project.

Reply:
Done. Unfortunately we can't do this for the backend stuff because it's using tox's `--parallel` mode. `make` also supports parallel but it's not as nice as tox in keeping the output of separate parallel jobs separate